test/e2e/apps/statefulset.go

```diff
@@ -1347,8 +1348,104 @@ var _ = SIGDescribe("StatefulSet", func() {
 			framework.ExpectNoError(err)
 		})
 	})
+
+	ginkgo.It("PVC should be recreated when pod is pending due to missing PVC [Feature:AutomaticPVCRecreation][Disruptive][Serial]", func() {
```

Comment:
I don't think you want the Feature tag? that will basically prevent it from ever being run.

@msau42 I'm not totally clear on when to use the feature tag, but I've already been bitten by adding it for autodelete, where the effect was to skip running it from the alpha cluster tests.

Reply:
@mattcary +1

Reply:
But the test requires some setup to be run (making sure the test pod will be scheduled on a node with local volumes) so that the pod will try to get rescheduled on the same node after being deleted. @msau42 suggested using the CSI mock driver [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/csi_mock_volume.go), but I am not sure how to use that to make a pod reschedule to the same node

Reply:
The Feature tag is used for alpha features or when something requires a special environment. @mattcary in order to run alpha feature tests in the alpha job, a PR to test-infra needs to be sent to opt-in the test in the alpha job here: https://github.com/kubernetes/test-infra/blob/72c610fe7e55ad8402966a161e8b66b41eb95912/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L375

Reply:
You can list nodes in the cluster and then add a node selector to the pod spec?

In that case this test should be able to run everywhere and a Feature tag shouldn't be necessary I think?

Reply:
Regarding this test case specifically, must it require local volumes or can you use Pod NodeAffinity to force the pod onto a single node? We want to test the Pod and PVC coordination under certain ordering of events, and while the sequence is most common when using local volumes, the solution is not local volume specific and can apply to any type of PVC.

We do have local-volume specific tests using in-tree local volume type, but it requires the ability to exec into the node to setup the directories. Using CSI mock driver avoids needing to exec into the host (although it still creates directories onto the node).

Reply:
Yes I just tried forcing the pod onto a single node using a nodeSelector and it reproduced the issue without having to use local volumes, I will make this change in the tests. Thank you for the suggestion!